test/hotspot/jtreg/compiler/integerArithmetic/DivisionByConstant.java

```diff
@@ -0,0 +1,226 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package compiler.integerArithmetic;
+
+import compiler.lib.ir_framework.*;
+import java.util.Random;
+
+import jdk.test.lib.Asserts;
+import jdk.test.lib.Utils;
+
+/*
+ * @test
+ * @bug 8282365
+ * @summary Test that Ideal transformations of division nodes provide correct
+ * result.
+ *
+ * @requires os.simpleArch=="x64"
+ * @library /test/lib /
+ * @run driver compiler.integerArithmetic.DivisionByConstant
+ */
+public class DivisionByConstant {
+    private static final int TRIALS = 10;
+    private static final int INVOCATIONS = 100;
+
+    private static final int I_DIV;
+    private static final int I_1;
+    private static final int I_2;
+    private static final long L_DIV;
+    private static final long L_1;;
+    private static final long L_2;
+
+    static {
+        int iDiv = 0;
+        int i1 = 0;
+        int i2 = 0;
+        long lDiv = 0;
+        long l1 = 0;
+        long l2 = 0;
+        try {
+            iDiv = Integer.parseUnsignedInt(System.getProperty("iDiv"));
+            i1 = Integer.parseUnsignedInt(System.getProperty("i1"));
+            i2 = Integer.parseUnsignedInt(System.getProperty("i2"));
+            lDiv = Long.parseUnsignedLong(System.getProperty("lDiv"));
+            l1 = Long.parseUnsignedLong(System.getProperty("l1"));
+            l2 = Long.parseUnsignedLong(System.getProperty("l2"));
+        } catch (Exception e) {}
+
+        I_DIV = iDiv;
+        I_1 = i1;
+        I_2 = i2;
+        L_DIV = lDiv;
+        L_1 = l1;
+        L_2 = l2;
+    }
+    private static final int I_LO = Math.min(I_1, I_2);
+    private static final int I_HI = Math.max(I_1, I_2);
+    private static final int I_UHI = I_1;
+    private static final long L_LO = Math.min(L_1, L_2);
+    private static final long L_HI = Math.max(L_1, L_2);
+    private static final long L_UHI = L_1;
+
+    public static void main(String[] args) {
+        Random r = Utils.getRandomInstance();
+        for (int i = 0; i < TRIALS; i++) {
+            String iDiv = Long.toUnsignedString(logRandom(r, Integer.SIZE));
+            String i1 = Long.toUnsignedString(logRandom(r, Integer.SIZE));
+            String i2 = Long.toUnsignedString(logRandom(r, Integer.SIZE));
+            String lDiv = Long.toUnsignedString(logRandom(r, Long.SIZE));
+            String l1 = Long.toUnsignedString(logRandom(r, Long.SIZE));
+            String l2 = Long.toUnsignedString(logRandom(r, Long.SIZE));
+
+            var test = new TestFramework(DivisionByConstant.class);
+            test.setDefaultWarmup(1);
+            test.addFlags("-DiDiv=" + iDiv, "-Di1=" + i1, "-Di2=" + i2,
+                    "-DlDiv=" + lDiv, "-Dl1=" + l1, "-Dl2=" + l2);
```

Comment:
Does this really change the constants in every run? Is the `static` block above that parses the arguments really called more than one time? Maybe the `TestFramework` does some magic here. Do you know how that works?

Reply:
The `TestFramework` spawns a separate VM for each `start` invocation so each will have its own set of `static final` variables. I have added a verification method to assert the validity of this approach.